src/assets/js/components/Tweet.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div>
+    <NavBar :isLoggedIn="isLoggedIn" />
+    <div v-if="isLoggedIn">
+      <div class="alert alert-danger" v-if="error">Error!</div>
+      <div class="alert alert-success" v-if="success">Success!</div>
+      <form @submit.prevent="tweet">
+        <input v-model="post" type="text" required />
+
+        <button type="submit">Tweet</button>
+      </form>
+    </div>
+    <div v-for="tweet in tweets" :key="tweet.id">
+      <span>{{ tweet.tweet }}</span>
+      <div v-if="isLoggedIn">
+        <button
+          v-if="tweet.user_id === parseInt(user_id)"
+          @click="remove(tweet.id)"
+        >
+          Delete
+        </button>
+        <button v-if="isFavorited(tweet)" @click="unfavorite(tweet.id)">
+          Unfavorite
+        </button>
+        <button v-else @click="favorite(tweet.id)">
+          Favorite
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import request from "../api/request";
+import NavBar from "./NavBar";
+export default {
+  name: "tweet",
+  components: {
+    NavBar
+  },
+  data() {
+    return {
+      tweets: [],
+      post: "",
+      user_id: 0,
+      error: false,
+      success: false,
+      isLoggedIn: false
+    };
+  },
+  async created() {
+    const user_id = sessionStorage.getItem("user_id");
```

Comment:
The default value of `user_id` is 0 (number type), but string type value is assigned in `async created()` because `sessionStorage.getItem()` behaves as shown below.

Therefore, in order to prevent unintended behavior, it would be better to use `JSON.parse()` when get data from `localStorage` or `sessionStorage`.

```js
sessionStorage.setItem("user_id", 0);
typeof sessionStorage.getItem("user_id"); // output: "string"

typeof JSON.parse(sessionStorage.getItem("user_id")); // output: "number"
```